src/unit-number/unit-number-input.tsx

```diff
@@ -211,18 +164,17 @@ export const UnitNumberInput = ({
          * Ignore inputs that exceed the respective max length
          */
 
-        const value = event.target.value.toLocaleUpperCase();
+        let value = event.target.value.toLocaleUpperCase();
+        if (!/.*[0-9A-Za-z]$/.test(value)) {
+            value = value.replace(/.$/, "");
```

Comment:
this regex will allow special characters at the front

for alphanumeric checks and replacement, try
```suggestion
       if (/[^0-9A-Za-z]/.test(value)) {
            value = value.replace(/[^0-9A-Za-z]/, "");
```